In [77]:
# Importing libraries
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.naive_bayes import GaussianNB
import matplotlib.pyplot as plt

In [55]:
# Importing dataset
df = pd.read_csv('Iris.csv', index_col=0)

In [56]:
# Looking for columns and rows
df.shape

(150, 5)

This shows that this dataset has 150 records(rows) and 6 features(columns).

In [57]:
# Making encoding dictionary
encoding_dict = {
    "Species" : {"Iris-setosa":0, "Iris-versicolor":1, "Iris-virginica":2}
}

In [58]:
# Replacing the categorical data to numerical data.
df = df.replace(encoding_dict)

In [59]:
# I/O data
X = df.iloc[:,:-1]
y = df.iloc[:,-1]

In [60]:
# Test dummy data.
X_dummy = X.iloc[5,:].values.reshape(1,-1)
y_dummy = y.iloc[5]

In [62]:
# Model initialization.
clf = GaussianNB()

In [63]:
# Tuple
result_tuple = tuple()

In [64]:
# Clf list
clfs = []

In [65]:
# Using for loop for getting train test data and model training and evaluation.
for i in np.arange(50,70,5):
    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.18, random_state=i)
    clf.fit(X_train, y_train)
    clfs.append(clf)

In [74]:
for clf in clfs:
    a_list = list(result_tuple)
    a_list.append(clf.predict(X_dummy))
    result_tuple = tuple(a_list)

/home/munsif/Notebook/JupyterEnv/lib/python3.8/site-packages/sklearn/base.py:450: UserWarning: X does not have valid feature names, but GaussianNB was fitted with feature names
  warnings.warn(
/home/munsif/Notebook/JupyterEnv/lib/python3.8/site-packages/sklearn/base.py:450: UserWarning: X does not have valid feature names, but GaussianNB was fitted with feature names
  warnings.warn(
/home/munsif/Notebook/JupyterEnv/lib/python3.8/site-packages/sklearn/base.py:450: UserWarning: X does not have valid feature names, but GaussianNB was fitted with feature names
  warnings.warn(
/home/munsif/Notebook/JupyterEnv/lib/python3.8/site-packages/sklearn/base.py:450: UserWarning: X does not have valid feature names, but GaussianNB was fitted with feature names
  warnings.warn(


In [75]:
print(result_tuple)

(array([], dtype=float64), array([0]), array([0]), array([0]), array([0]))


In [80]:
print(y_dummy)

0


The actual value is 0 and all the other classifiers are also giving 0 as output result.